cannot upgrade snapshot release of IDEasy

### Actual behavior

```
$ ide upgrade --mode=snapshot
Trying to download ide-cli-*!-linux-x64.tar.gz from https://repo1.maven.org/maven2/com/devonfw/tools/IDEasy/ide-cli/ide-cli-*!-linux-x64.tar.gz
An unexpected error occurred!
We are sorry for the inconvenience.
Please check the error below, resolve it and try again.
If the error is not on your end (network connectivity, lack of permissions, etc.) please file a bug:
https://github.com/devonfw/IDEasy/issues/new?assignees=&labels=bug&projects=&template=bug.md&title=IllegalStateException%3A+Failed+to+download+file+from+URL+https%3A%2F%2Frepo1.maven.org%2Fmaven2%2Fcom%2Fdevonfw%2Ftools%2FIDEasy%2Fide-cli%2Fide-cli-*%21-linux-x64.tar.gz+to+%2Fhome%2Fhohwille%2Fprojects%2F_ide%2Ftmp%2FDownloads%2Fide-cli-*%21-linux-x64.tar.gz
java.lang.IllegalStateException: Failed to download file from URL https://repo1.maven.org/maven2/com/devonfw/tools/IDEasy/ide-cli/ide-cli-*!-linux-x64.tar.gz to /home/hohwille/projects/_ide/tmp/Downloads/ide-cli-*!-linux-x64.tar.gz
        at com.devonfw.tools.ide.io.FileAccessImpl.downloadViaHttp(FileAccessImpl.java:140)
        at com.devonfw.tools.ide.io.FileAccessImpl.download(FileAccessImpl.java:105)
        at com.devonfw.tools.ide.tool.repository.AbstractToolRepository.download(AbstractToolRepository.java:197)
        at com.devonfw.tools.ide.tool.repository.MvnRepository.getDownloadedArtifact(MvnRepository.java:150)
        at com.devonfw.tools.ide.tool.repository.MvnRepository.download(MvnRepository.java:247)
        at com.devonfw.tools.ide.tool.mvn.MvnBasedLocalToolCommandlet.downloadTool(MvnBasedLocalToolCommandlet.java:55)
        at com.devonfw.tools.ide.tool.LocalToolCommandlet.performToolInstallation(LocalToolCommandlet.java:264)
        at com.devonfw.tools.ide.tool.LocalToolCommandlet.installTool(LocalToolCommandlet.java:243)
        at com.devonfw.tools.ide.tool.LocalToolCommandlet.installTool(LocalToolCommandlet.java:187)
        at com.devonfw.tools.ide.tool.LocalToolCommandlet.doInstallStep(LocalToolCommandlet.java:100)
        at com.devonfw.tools.ide.tool.LocalToolCommandlet.install(LocalToolCommandlet.java:90)
        at com.devonfw.tools.ide.tool.ToolCommandlet.install(ToolCommandlet.java:266)
        at com.devonfw.tools.ide.tool.IdeasyCommandlet.install(IdeasyCommandlet.java:122)
        at com.devonfw.tools.ide.commandlet.UpgradeCommandlet.run(UpgradeCommandlet.java:43)
        at com.devonfw.tools.ide.context.AbstractIdeContext.applyAndRun(AbstractIdeContext.java:1120)
        at com.devonfw.tools.ide.context.AbstractIdeContext.run(AbstractIdeContext.java:1030)
        at com.devonfw.tools.ide.cli.Ideasy.runOrThrow(Ideasy.java:95)
        at com.devonfw.tools.ide.cli.Ideasy.run(Ideasy.java:59)
        at com.devonfw.tools.ide.cli.Ideasy.main(Ideasy.java:133)
        at java.base@23.0.2/java.lang.invoke.LambdaForm$DMH/sa346b79c.invokeStaticInit(LambdaForm$DMH)
Caused by: java.lang.IllegalStateException: Failed to stream response body from url: https://repo1.maven.org/maven2/com/devonfw/tools/IDEasy/ide-cli/ide-cli-*!-linux-x64.tar.gz
        at com.devonfw.tools.ide.io.HttpDownloader.httpGet(HttpDownloader.java:89)
        at com.devonfw.tools.ide.io.FileAccessImpl.downloadWithHttpVersion(FileAccessImpl.java:151)
        at com.devonfw.tools.ide.io.FileAccessImpl.downloadViaHttp(FileAccessImpl.java:125)
        ... 19 more
Caused by: java.lang.IllegalStateException: Download failed with status code 404
        at com.devonfw.tools.ide.io.HttpDownloader.httpGet(HttpDownloader.java:86)
        ... 21 more


Error: IDEasy failed with exit code 255
```

### Reproduce

`ide upgrade --mode=snapshot`

### Expected behavior

upgrade should succeed and install latest snapshot release.

### IDEasy status

```bash
IDE_ROOT is set to ~/projects
You are not inside an IDE project: ~
You are online.
Your version of IDEasy is 2025.09.001-09_22_02-SNAPSHOT.
You are using a SNAPSHOT version of IDEasy. For stability consider switching to a stable release via 'ide upgrade --mode=stable'
Your version of IDEasy is 2025.09.001-09_22_02-SNAPSHOT but version 2025.07.001-20250723.025205-3 is available. Please run the following command to upgrade to the latest version:
ide upgrade
Your operating system is linux(5.15.133.1-microsoft-standard-WSL2)@x64 [Linux@amd64]
Successfully completed ide (status)
```

### Related/Dependent issues

_No response_

### Comments/Hints

_No response_

This is what I see on Linux:
```
$ ide upgrade --mode=snapshot
Trying to download ide-cli-2025.09.001-20250929.135438-10-linux-x64.tar.gz from https://central.sonatype.com/repository/maven-snapshots/com/devonfw/tools/IDEasy/ide-cli/2025.09.001-SNAPSHOT/ide-cli-2025.09.001-20250929.135438-10-linux-x64.tar.gz
Downloading 100% │██████████████████████████████████████████████████████████████│ 23/23MiB (0:00:03 / 0:00:00) 7.8MiB/s
Trying to download ide-cli-2025.09.001-20250929.135438-10-linux-x64.tar.gz.md5 from https://central.sonatype.com/repository/maven-snapshots/com/devonfw/tools/IDEasy/ide-cli/2025.09.001-SNAPSHOT/ide-cli-2025.09.001-20250929.135438-10-linux-x64.tar.gz.md5
Trying to download ide-cli-2025.09.001-20250929.135438-10-linux-x64.tar.gz.sha1 from https://central.sonatype.com/repository/maven-snapshots/com/devonfw/tools/IDEasy/ide-cli/2025.09.001-SNAPSHOT/ide-cli-2025.09.001-20250929.135438-10-linux-x64.tar.gz.sha1
MD5 checksum 811b524623b67020ed1bde0da2c90ce7 is correct.
SHA1 checksum 2cccd7c18014bfb967e66efc7bde5c5c40b22a55 is correct.
Extracting TAR file /home/hohwille/.m2/repository/com/devonfw/tools/IDEasy/ide-cli/2025.09.001-SNAPSHOT/ide-cli-2025.09.001-20250929.135438-10-linux-x64.tar.gz to /home/hohwille/projects/_ide/tmp/extract-ide-cli-2025.09.001-20250929.135438-10-linux-x64.tar.gz
Extracting 100% │█████████████████████████████████████████████████████████████████│ 23/23MiB (0:00:00 / 0:00:00) ?MiB/s
Successfully installed ideasy in version 2025.09.001-20250929.135438-10 replacing previous version 2025.08.001
It is recommended to run 'ide update' on your IDEasy projects now.
```
and then
```
hohwille@CE49454:~/projects/_ide$ ide upgrade --mode=snapshot
Trying to download ide-cli-*!-linux-x64.tar.gz from https://repo1.maven.org/maven2/com/devonfw/tools/IDEasy/ide-cli/ide-cli-*!-linux-x64.tar.gz
An unexpected error occurred!
We are sorry for the inconvenience.
Please check the error below, resolve it and try again.
If the error is not on your end (network connectivity, lack of permissions, etc.) please file a bug:
https://github.com/devonfw/IDEasy/issues/new?assignees=&labels=bug&projects=&template=bug.md&title=IllegalStateException%3A+Failed+to+download+file+from+URL+https%3A%2F%2Frepo1.maven.org%2Fmaven2%2Fcom%2Fdevonfw%2Ftools%2FIDEasy%2Fide-cli%2Fide-cli-*%21-linux-x64.tar.gz+to+%2Fhome%2Fhohwille%2Fprojects%2F_ide%2Ftmp%2FDownloads%2Fide-cli-*%21-linux-x64.tar.gz
java.lang.IllegalStateException: Failed to download file from URL https://repo1.maven.org/maven2/com/devonfw/tools/IDEasy/ide-cli/ide-cli-*!-linux-x64.tar.gz to /home/hohwille/projects/_ide/tmp/Downloads/ide-cli-*!-linux-x64.tar.gz
        at com.devonfw.tools.ide.io.FileAccessImpl.downloadViaHttp(FileAccessImpl.java:140)
        at com.devonfw.tools.ide.io.FileAccessImpl.download(FileAccessImpl.java:105)
        at com.devonfw.tools.ide.tool.repository.AbstractToolRepository.download(AbstractToolRepository.java:197)
        at com.devonfw.tools.ide.tool.repository.MvnRepository.getDownloadedArtifact(MvnRepository.java:150)
        at com.devonfw.tools.ide.tool.repository.MvnRepository.download(MvnRepository.java:247)
        at com.devonfw.tools.ide.tool.mvn.MvnBasedLocalToolCommandlet.downloadTool(MvnBasedLocalToolCommandlet.java:55)
        at com.devonfw.tools.ide.tool.LocalToolCommandlet.performToolInstallation(LocalToolCommandlet.java:264)
        at com.devonfw.tools.ide.tool.LocalToolCommandlet.installTool(LocalToolCommandlet.java:243)
        at com.devonfw.tools.ide.tool.LocalToolCommandlet.installTool(LocalToolCommandlet.java:187)
        at com.devonfw.tools.ide.tool.LocalToolCommandlet.doInstallStep(LocalToolCommandlet.java:100)
        at com.devonfw.tools.ide.tool.LocalToolCommandlet.install(LocalToolCommandlet.java:90)
        at com.devonfw.tools.ide.tool.ToolCommandlet.install(ToolCommandlet.java:266)
        at com.devonfw.tools.ide.tool.IdeasyCommandlet.install(IdeasyCommandlet.java:122)
        at com.devonfw.tools.ide.commandlet.UpgradeCommandlet.run(UpgradeCommandlet.java:43)
        at com.devonfw.tools.ide.context.AbstractIdeContext.applyAndRun(AbstractIdeContext.java:1120)
        at com.devonfw.tools.ide.context.AbstractIdeContext.run(AbstractIdeContext.java:1030)
        at com.devonfw.tools.ide.cli.Ideasy.runOrThrow(Ideasy.java:95)
        at com.devonfw.tools.ide.cli.Ideasy.run(Ideasy.java:59)
        at com.devonfw.tools.ide.cli.Ideasy.main(Ideasy.java:133)
        at java.base@23.0.2/java.lang.invoke.LambdaForm$DMH/sa346b79c.invokeStaticInit(LambdaForm$DMH)
Caused by: java.lang.IllegalStateException: Failed to stream response body from url: https://repo1.maven.org/maven2/com/devonfw/tools/IDEasy/ide-cli/ide-cli-*!-linux-x64.tar.gz
        at com.devonfw.tools.ide.io.HttpDownloader.httpGet(HttpDownloader.java:89)
        at com.devonfw.tools.ide.io.FileAccessImpl.downloadWithHttpVersion(FileAccessImpl.java:151)
        at com.devonfw.tools.ide.io.FileAccessImpl.downloadViaHttp(FileAccessImpl.java:125)
        ... 19 more
Caused by: java.lang.IllegalStateException: Download failed with status code 404
        at com.devonfw.tools.ide.io.HttpDownloader.httpGet(HttpDownloader.java:86)
        ... 21 more


Error: IDEasy failed with exit code 255
```

--- a/CHANGELOG.adoc
+++ b/CHANGELOG.adoc
@@ -13,6 +13,7 @@ Release with new features and bugfixes:
 * https://github.com/devonfw/IDEasy/issues/1221[#1221]: Improve Windows terminal integration on installation
 * https://github.com/devonfw/IDEasy/issues/1460[#1460]: Prevent strange ansi-codes in logs
 * https://github.com/devonfw/IDEasy/issues/1484[#1484]: Broken error handling: Couldn't uninstall node.
+* https://github.com/devonfw/IDEasy/issues/1515[#1515]: cannot upgrade snapshot release of IDEasy
 
 The full list of changes for this release can be found in https://github.com/devonfw/IDEasy/milestone/34?closed=1[milestone 2025.09.001].
 

--- a/cli/src/main/java/com/devonfw/tools/ide/tool/mvn/MvnBasedLocalToolCommandlet.java
+++ b/cli/src/main/java/com/devonfw/tools/ide/tool/mvn/MvnBasedLocalToolCommandlet.java
@@ -51,6 +51,7 @@ public abstract class MvnBasedLocalToolCommandlet extends LocalToolCommandlet {
 
     MvnRepository mvnRepository = this.context.getMvnRepository();
     MvnArtifact mavenArtifact = getArtifact(edition);
+    mavenArtifact = mavenArtifact.withVersion(resolvedVersion.toString());
     MvnArtifactMetadata mavenArtifactMetadata = mvnRepository.getMetadata(mavenArtifact, this.tool, edition);
     return mvnRepository.download(mavenArtifactMetadata);
   }

--- a/cli/src/main/java/com/devonfw/tools/ide/tool/repository/MvnRepository.java
+++ b/cli/src/main/java/com/devonfw/tools/ide/tool/repository/MvnRepository.java
@@ -209,7 +209,7 @@ public final class MvnRepository extends ArtifactToolRepository<MvnArtifact, Mvn
     NodeList versionsChildren = versions.getElementsByTagName("version");
     int length = versionsChildren.getLength();
     List<VersionIdentifier> versionList = new ArrayList<>(length);
-    for (int i = 0; i < length; i++) {
+    for (int i = length - 1; i >= 0; i--) {
       versionList.add(VersionIdentifier.of(versionsChildren.item(i).getTextContent()));
     }
     return versionList;

